File: model/src/main/java/io/github/runtimemodels/chazm/event/PmfEventFactory.java
package io.github.runtimemodels.chazm.event;

import io.github.runtimemodels.chazm.entity.Pmf;

/**
 * The {@linkplain PmfEventFactory} interface defines the API for constructing {@linkplain PmfEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface PmfEventFactory {

    /**
     * Constructs an {@linkplain PmfEvent}.
     *
     * @param category the {@linkplain EventCategory}.
     * @param pmf      the {@linkplain Pmf}.
     * @return a {@linkplain PmfEvent}.
     */
    PmfEvent build(EventCategory category, Pmf pmf);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/PmfEventFactory.kt
package runtimemodels.chazm.model.entity

import runtimemodels.chazm.api.entity.Pmf
import runtimemodels.chazm.model.event.EventType

/**
 * The [PmfEventFactory] interface defines the API for constructing [PmfEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface PmfEventFactory {
    /**
     * Constructs an [PmfEvent].
     *
     * @param category the [EventType].
     * @param pmf      the [Pmf].
     * @return a [PmfEvent].
     */
    fun build(category: EventType, pmf: Pmf): PmfEvent
}
